libs/ui/src/modals/screens/d-app-connection-confirmation/d-app-connection-confirmation.tsx

```diff
@@ -51,18 +52,19 @@ export const DAppConnectionConfirmation: FC = () => {
 
   const { params } = useRoute<RouteProp<ScreensParamList, ScreensEnum.DAppConnectionConfirmation>>();
 
-  useClosePopup(params.messageId);
+  useClosePopup(params.messageId, params.dAppInfo.origin);
 
   const gasBalance = getFormattedBalance(balance.data, decimals);
 
   const sendMessage = () => {
     dispatch(connectDAppAction({ dAppInfo: params.dAppInfo, accountPublicKeyHash: selectedAccountPublicKeyHash }));
-    sendNotificationToDApp('oko_accountsChanged', [selectedAccountPublicKeyHash]);
-    sendResponseToDAppAndClosePopup(params.messageId, [selectedAccountPublicKeyHash]);
+    sendNotificationToDApp('oko_accountsChanged', [selectedAccountPublicKeyHash], params.dAppInfo.origin);
+    sendResponseToDAppAndClosePopup(params.messageId, [selectedAccountPublicKeyHash], params.dAppInfo.origin);
```

Comment:
let's make single params order for all messages: origin, messageId, paramsToSend